### PR TITLE
Move `pkg/kubelet/cri/remote` to `cri-client`

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Follow-up on https://github.com/kubernetes/kubernetes/pull/123797

Refers to https://github.com/kubernetes/org/issues/4805 and https://github.com/kubernetes/test-infra/pull/32537

#### Which issue(s) this PR fixes:

Requires https://github.com/kubernetes/kubernetes/pull/123795 and https://github.com/kubernetes/kubernetes/pull/123796

#### Does this PR introduce a user-facing change?

```release-note
Moved remote CRI implementation from kubelet to `k8s.io/cri-client` repository.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes/sig-node-pr-reviews PTAL. I think it's better to get the code-reorg done before the actual feature development starts for the upcoming release cycle.

File: staging/src/k8s.io/cri-client/pkg/errors/errors.go
Comment (SergeyKanzhelev) [inline staging/src/k8s.io/cri-client/pkg/errors/errors.go:20]: there is a TODO in this file suggesting we can start using grpc errors: `TODO: this error type can likely be removed when we support CRI errors.` I *think* it is a leftover after dockershim. Did you by chance have a chance to look at how expensive will it be to eliminate the need of this type before the migration?

File: staging/src/k8s.io/cri-client/pkg/remote_image.go
Comment (SergeyKanzhelev) [inline staging/src/k8s.io/cri-client/pkg/remote_image.go:40]: Are we keeping the error logging from inside the library code (`klog.ErrorS`)? Typically libraries returns descriptive error object and let caller decide how to log it.

File: staging/src/k8s.io/cri-client/pkg/errors/errors.go
Comment (saschagrunert) [reply staging/src/k8s.io/cri-client/pkg/errors/errors.go:20]: Sure, following-up in https://github.com/kubernetes/kubernetes/pull/124739

File: staging/src/k8s.io/cri-client/pkg/remote_image.go
Comment (saschagrunert) [reply staging/src/k8s.io/cri-client/pkg/remote_image.go:40]: The problem is that klog.Logger is not capable of logging structured via `InfoS` (only `Info`). Is it okay to drop that requirement? Then I'd say we can pass a `*klog.Logger` to the services which can be `nil` as well: https://github.com/kubernetes/kubernetes/pull/124741

Comment (saschagrunert) [reply staging/src/k8s.io/cri-client/pkg/remote_image.go:40]: Solved.

**Comment by saschagrunert** [comment]
Ready for review again. PTAL @kubernetes/sig-node-pr-reviews

**Comment by dims** [comment]
@saschagrunert please assign this to me after you get some lgtm(s) from sig-node reviewers/approvers, thanks!

**Comment by saschagrunert** [comment]
@SergeyKanzhelev @haircommander @mrunalp is this ready?

**Comment by haircommander** [comment]
the code moving pieces LGTM

one thought I just had: could it make more sense to put this in a `k8s.io/cri-api/pkg/client` package vs the totally separate `k8s.io/cri-api` package?

**Comment by saschagrunert** [comment]
> the code moving pieces LGTM
> 
> one thought I just had: could it make more sense to put this in a `k8s.io/cri-api/pkg/client` package vs the totally separate `k8s.io/cri-api` package?

@haircommander we already added the new `cri-client` staging repo to the architecture for the purpose of using the API separately. For example CRI-O would not use `cri-client`, but `cri-api`. cri-tools would use `cri-client` as well as the API.

**Comment by haircommander** [comment]
works for me
/lgtm 

nice cleanup here!

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

thank you for addressing comments!

**Pull request merged**
